runtime/calamari/src/migrations/asset_id.rs

```diff
@@ -0,0 +1,680 @@
+type NewAssetId = u128;
+
+pub mod old {
+    use crate::migrations::asset_id::OldAssetId;
+    use frame_support::{storage_alias, Blake2_128Concat};
+
+    #[storage_alias]
+    pub type Account<T: frame_system::Config<I>, I: 'static = ()> = StorageDoubleMap<
+        Assets,
+        Blake2_128Concat,
+        OldAssetId,
+        Blake2_128Concat,
+        <T as frame_system::Config>::AccountId,
+        super::AssetAccountOf<T, I>,
+    >;
+}
+
+#[storage_alias]
+type Account<T: frame_system::Config<I>, I: 'static = ()> = StorageDoubleMap<
+    Assets,
+    Blake2_128Concat,
+    NewAssetId,
+    Blake2_128Concat,
+    <T as frame_system::Config>::AccountId,
+    AssetAccountOf<T, I>,
+>;
+
+type AssetMapKVP<T> = (
+    OldAssetId,
+    pallet_assets::AssetDetails<
+        <T as pallet_assets::Config>::Balance,
+        <T as frame_system::Config>::AccountId,
+        DepositBalanceOf<T>,
+    >,
+);
+
+type MetadataMapKVP<T> = (
+    OldAssetId,
+    pallet_assets::AssetMetadata<
+        DepositBalanceOf<T>,
+        BoundedVec<u8, <T as pallet_assets::Config>::StringLimit>,
+    >,
+);
+
+#[derive(Clone, Debug, Decode, Encode, Eq, Hash, Ord, PartialEq, PartialOrd, TypeInfo)]
+pub struct OldAssetRegistrarMetadata {
+    pub name: Vec<u8>,
+    pub symbol: Vec<u8>,
+    pub decimals: u8,
+    pub evm_address: Option<H160>,
+    pub is_frozen: bool,
+    pub min_balance: Balance,
+    pub is_sufficient: bool,
+}
+
+pub struct AssetIdMigration<T, M, A>(PhantomData<T>, PhantomData<M>, PhantomData<A>);
+impl<T, M, A> OnRuntimeUpgrade for AssetIdMigration<T, M, A>
+where
+    T: pallet_asset_manager::Config + pallet_assets::Config,
+    M: GetStorageVersion + PalletInfoAccess,
+    A: GetStorageVersion + PalletInfoAccess,
+{
+    fn on_runtime_upgrade() -> Weight {
+        let mut num_reads = 0;
+        let mut num_writes = 0;
+
+        let asset_manager_storage_version = <M as GetStorageVersion>::on_chain_storage_version();
+        let assets_storage_version = <A as GetStorageVersion>::on_chain_storage_version();
+        num_reads += 2;
+        if asset_manager_storage_version != 1 || assets_storage_version != 0 {
+            return T::DbWeight::get().reads(num_reads as Weight);
+        }
+
+        // AssetIdLocation
+
+        let pallet_prefix: &[u8] = b"AssetManager";
+        let storage_item_prefix: &[u8] = b"AssetIdLocation";
+        let stored_data: Vec<_> = storage_key_iter::<OldAssetId, AssetLocation, Blake2_128Concat>(
+            pallet_prefix,
+            storage_item_prefix,
+        )
+        .drain()
+        .collect();
+        for (old_key, value) in stored_data {
+            let new_key: NewAssetId = old_key as NewAssetId;
+            put_storage_value(
+                pallet_prefix,
+                storage_item_prefix,
+                &Blake2_128Concat::hash(&new_key.encode()),
+                value,
+            );
+            num_reads += 1;
+            num_writes += 1;
+        }
+
```

Comment:
also progress reports like log::info!("Migrated {num_of_locations} AssetIdLocations")

You can get rid of the in-code comments if you just `log` the stages, that gives more info at runtime (especially if - god forbid - the migration fails on mainnet)